src/App.tsx

```diff
@@ -1,223 +1,112 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useCallback,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
+import { getTodos } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { NewTodo } from './components/NewTodo';
+import { TodosFilter } from './components/TodoFilter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoStatus } from './types/TodoStatus';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
```

Comment:
no need to create a new state variable if you can get it from another state variable
just create variable a common `visibleTodos`